tests/general/container_test.go

```diff
@@ -115,3 +115,65 @@ func TestSpecifiedContainerConfigDefaultsToCmdLineArgIfEnvVarConflict(t *testing
 		return false
 	}, 5*time.Second, time.Second)
 }
+
+func TestConfigYamlEnvVarUsingLogs(t *testing.T) {
+	image := os.Getenv("SPLUNK_OTEL_COLLECTOR_IMAGE")
+	if strings.TrimSpace(image) == "" {
+		t.Skipf("skipping container-only test")
+	}
+
+	// configYamlStr is the single-line version of the config YAML below.
+	//
+	//receivers:
+	//  hostmetrics:
+	//    collection_interval: 1s
+	//    scrapers:
+	//      cpu:
+	//exporters:
+	//  logging:
+	//    logLevel: debug
+	//service:
+	//  pipelines:
+	//    metrics:
+	//      receivers: [hostmetrics]
+	//      exporters: [logging]
+	//
+	configYamlStr := "receivers:\n  hostmetrics:\n    collection_interval: 1s\n    scrapers:\n      cpu:\nexporters:\n  logging:\n    logLevel: debug\nservice:\n  pipelines:\n    metrics:\n      receivers: [hostmetrics]\n      exporters: [logging]"
+	configYamlEnv := map[string]string{"SPLUNK_CONFIG_YAML": configYamlStr}
+
+	logCore, logs := observer.New(zap.DebugLevel)
+	logger := zap.New(logCore)
+
+	collector, err := testutils.NewCollectorContainer().
+		WithImage(image).
+		WithEnv(configYamlEnv).
```

Comment:
I could be wrong but if you don't set `SPLUNK_CONFIG: ""` and are seeing the that the yaml is being used this suggests something isn't going as expected in your changes: https://github.com/signalfx/splunk-otel-collector/blob/main/tests/testutils/collector_container.go#L192

Reply:
So, the execution path never gets to line 192 in this case because collector.ConfigPath is zero here https://github.com/signalfx/splunk-otel-collector/blob/main/tests/testutils/collector_container.go#L162

Reply:
@rmfitzpatrick does my response make sense?